### PR TITLE
update recreate service api with service order table.

### LINKED ISSUE #2029: remove SERVICE_RECREATE table and consolidate into SERVICE_ORDER
- [ ] delete SERVICE_RECREATE table. 
-  Extend SERVICE_ORDER table with WORKFLOW_ID table -> This must be the activiti instance ID. Use this ID to fetch the SERVICE_ID from the activati execution context. 
- [ ] Extend ServiceOrderType enum with 'RECREATE' value. 
- [ ] Create SERVICE_ORDER entry as part of the API request and return the ServiceOrder Object. 
- [ ]  As part of callbacks and exceptions, set status directly to deploy_service and service_order tables. 
- [ ] Keep only recreateService method and delete all other methods. 
- [ ] Test task retry handing for failures.

### PR DESCRIPTION
**Pull request by Alice1319**
fixes #2029 

recreate successfully:
image

recreate-destroy-retry:
image

recreate-deploy-retry:
image

### CODE REVIEW

**Comment by WangLiNaruto** [comment]
LGTM

**Comment by wuzhuanhong** [comment]
LGTM

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceRecreateApi.java [hunk 1]
```diff
@@ -112,74 +113,51 @@ public UUID recreateService(@Valid @PathVariable("serviceId") String serviceId)
...
+            ServiceOrderEntity recreateOrderEntity) {
         Map<String, Object> variable = new HashMap<>();
-        variable.put(RecreateConstants.ID, deployedService.getDeployRequest().getServiceId());
+        variable.put(RecreateConstants.SERVICE_ID, deployedService.getId());
+        variable.put(RecreateConstants.RECREATE_ORDER_ID, recreateOrderEntity.getOrderId());
         variable.put(RecreateConstants.RECREATE_REQUEST, deployedService.getDeployRequest());
-        variable.put(RecreateConstants.USER_ID, userId);
+        variable.put(RecreateConstants.USER_ID, recreateOrderEntity.getUserId());
+        variable.put(RecreateConstants.DEPLOY_RETRY_NUM, 0);
+        variable.put(RecreateConstants.DESTROY_RETRY_NUM, 0);
```
Comment (baixinsui) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceRecreateApi.java:160]: I think we should add a constant named `RERATE_ERROR_MESSAGE` RecreateConstants, which uesd to store the error_message in the filed deploy order or destory order.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/consts/RecreateConstants.java [hunk 2]
```diff
@@ -12,7 +12,8 @@
 public class RecreateConstants {
 
     public static final String PROCESS_KEY = "recreate";
-    public static final String ID = "id";
+    public static final String SERVICE_ID = "serviceId";
```
Comment (baixinsui) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/consts/RecreateConstants.java:15]: we can add a constant named RECREATE_ERROR_MESSAGE, used to store the  error message when the order task failed

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDeploymentResult.java [hunk 3]
```diff
@@ -50,42 +50,40 @@ public void execute(DelegateExecution execution) {
...
+            log.info("Deployment step completed for recreate order workflow with id {}.",
+                    processInstanceId);
         } else {
-            recreateService.updateServiceRecreateStatus(serviceRecreateEntity,
-                    RecreateStatus.DEPLOY_FAILED, OffsetDateTime.now());
             runtimeService.setVariable(processInstanceId, RecreateConstants.IS_DEPLOY_SUCCESS,
                     false);
-            int deployRetryNum = updateDeployRetryNum(processInstanceId, variables);
+            int deployRetryNum = getDeployRetryNum(variables);
+            runtimeService.setVariable(processInstanceId, RecreateConstants.DEPLOY_RETRY_NUM,
```
Comment (baixinsui) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDeploymentResult.java:70]: Get the error message from the ServiceDeploymentEntity and set it as the value of  'RECREATE_ERROR_MESSAGE' into variables of the processIntanceId

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDeploymentResult.java [hunk 4]
```diff
@@ -50,42 +50,40 @@ public void execute(DelegateExecution execution) {
...
+                    deployRetryNum + 1);
+            log.info("Process instance： {} retry to deploy task during recreation workflow. "
+                    + "RetryCount:{}", processInstanceId, deployRetryNum);
             if (deployRetryNum >= 1) {
                 String userId = (String) variables.get(RecreateConstants.USER_ID);
-                runtimeService.setVariable(processInstanceId, RecreateConstants.ASSIGNEE,
-                        userId);
+                runtimeService.setVariable(processInstanceId, RecreateConstants.ASSIGNEE, userId);
+                serviceOrderManager.completeOrderProgress(recreateOrderId, TaskStatus.FAILED,
+                        null);
```
Comment (baixinsui) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDeploymentResult.java:78]: Use the error message to replace null value when calling serviceOrderManager.completeOrderProgress(recreateOrderId, TaskStatus.FAILED,
                        null);

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDestroyResult.java [hunk 5]
```diff
@@ -30,64 +30,66 @@
...
 
         if (isDestroySuccess) {
-            recreateService.updateServiceRecreateStatus(serviceRecreateEntity,
-                    RecreateStatus.DESTROY_COMPLETED, OffsetDateTime.now());
             runtimeService.setVariable(processInstanceId, RecreateConstants.IS_DESTROY_SUCCESS,
                     true);
             runtimeService.setVariable(processInstanceId, RecreateConstants.DEPLOY_RETRY_NUM, 0);
-            log.info("destroy step completed for recreate order {}", processInstanceId);
+            try {
+                Thread.sleep(60 * 1000);
```
Comment (baixinsui) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDestroyResult.java:66]: This code do any effects?

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDestroyResult.java [hunk 6]
```diff
@@ -30,64 +30,66 @@
...
-            recreateService.updateServiceRecreateStatus(serviceRecreateEntity,
-                    RecreateStatus.DESTROY_FAILED, OffsetDateTime.now());
-
             runtimeService.setVariable(processInstanceId, RecreateConstants.IS_DESTROY_SUCCESS,
                     false);
-            int destroyRetryNum = updateDestroyRetryNum(processInstanceId, variables);
+            int destroyRetryNum = getDestroyRetryNum(variables);
+            runtimeService.setVariable(processInstanceId, RecreateConstants.DESTROY_RETRY_NUM,
+                    destroyRetryNum + 1);
+            log.info("Process instance： {} retry to destroy task during recreation workflow. "
```
Comment (baixinsui) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDestroyResult.java:78]: Get the error message from the ServiceDeploymentEntity and set it as the value of 'RECREATE_ERROR_MESSAGE' into variables of the processIntanceId

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDestroyResult.java [hunk 7]
```diff
@@ -30,64 +30,66 @@
...
+                    + "RetryCount:{}", processInstanceId, destroyRetryNum);
             if (destroyRetryNum >= 1) {
                 String userId = (String) variables.get(RecreateConstants.USER_ID);
                 runtimeService.setVariable(processInstanceId, RecreateConstants.ASSIGNEE, userId);
+                serviceOrderManager.completeOrderProgress(recreateOrderId,
+                        TaskStatus.FAILED, null);
```
Comment (baixinsui) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDestroyResult.java:84]: Use the error message to replace null value when calling serviceOrderManager.completeOrderProgress(recreateOrderId, TaskStatus.FAILED,
null);

**Comment by baixinsui** [comment]
@Alice1319 There is a samll error need to be fixed. When the recreate workflow failed, in the data of parent recreated order,  the error_message should not be null. Please updated the code as thees comments.

**Comment by yuanxiaojun987** [comment]
LGTM

**Comment by Alice1319** [comment]
> @Alice1319 There is a samll error need to be fixed. When the recreate workflow failed, in the data of parent recreated order, the error_message should not be null. Please updated the code as thees comments.

good idea, let me check this.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDestroyResult.java (hunk 5 above)
Comment (Alice1319) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDestroyResult.java:66]: yes, let me remove this test case.

**Comment by profoundwu** [comment]
LGTM

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/consts/RecreateConstants.java (hunk 2 above)
Comment (Alice1319) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/consts/RecreateConstants.java:15]: fixed
like we discussed, use deploymentEntityHandler.getServiceDeploymentEntity(serviceId) to query resultMessage.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDeploymentResult.java (hunk 3 above)
Comment (Alice1319) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDeploymentResult.java]: fixed
like we discussed, use deploymentEntityHandler.getServiceDeploymentEntity(serviceId) to query resultMessage.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDestroyResult.java (hunk 7 above)
Comment (Alice1319) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDestroyResult.java]: fixed
like we discussed, use deploymentEntityHandler.getServiceDeploymentEntity(serviceId) to query resultMessage.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**